falcon/request.py

```diff
@@ -1119,18 +1128,25 @@ def _parse_rfc_forwarded(self):
         Returns:
             list: addresses derived from "for" parameters.
         """
+
         addr = []
+
         for forwarded in self.env['HTTP_FORWARDED'].split(','):
             for param in forwarded.split(';'):
-                param = param.strip().split('=', 1)
-                if len(param) == 1:
+                # PERF(kgriffs): Partition() is faster than split().
+                key, _, val = param.strip().partition('=')
+                if not val:
+                    # NOTE(kgriffs): The '=' separator was not found or
+                    # it was, but the value was missing.
```

Comment:
This is slightly different from previous behavior, which would return an empty string in the case of an empty value. In this case, the field is malformed and behavior is ostensibly undefined, so let's just be consistent and always skip it when there is a problem.